gtsam/discrete/tests/testDecisionTree.cpp

```diff
@@ -40,7 +40,19 @@ void dot(const T&f, const string& filename) {
 
 #define DOT(x)(dot(x,#x))
 
-struct Crazy { int a; double b; };
+struct Crazy {
+  int a;
+  double b;
+
+  bool equals(const Crazy& other, double tol = 1e-12) const {
```

Comment:
why?

Reply:
This is needed so that the compiler doesn't complain about the default comparator method. Technically we don't need this, just the == operator overload.

Reply:
I think this points to an issue, same as equals_star issue: the compare function we pass in should not have a tol! That is a detail for the derived classes. For a simple type like this, the default compare should just be ==. If you want a tol, you need to construct a lambda, where tol is passed in as context.